kpi/deployment_backends/mock_backend.py

```diff
@@ -23,59 +25,86 @@
 from kpi.interfaces.sync_backend_media import SyncBackendMediaInterface
 from kpi.models.asset_file import AssetFile
 from kpi.utils.mongo_helper import MongoHelper, drop_mock_only
+from kpi.utils.xml import edit_submission_xml
 from .base_backend import BaseDeploymentBackend
+from ..exceptions import KobocatBulkUpdateSubmissionsClientException
 
 
 class MockDeploymentBackend(BaseDeploymentBackend):
     """
     Only used for unit testing and interface testing.
     """
 
+    PROTECTED_XML_FIELDS = [
+        '__version__',
+        'formhub',
+        'meta',
+    ]
+
     def bulk_assign_mapped_perms(self):
         pass
 
     def bulk_update_submissions(
         self, data: dict, user: 'auth.User'
     ) -> dict:
-
         submission_ids = self.validate_access_with_partial_perms(
             user=user,
             perm=PERM_CHANGE_SUBMISSIONS,
             submission_ids=data['submission_ids'],
             query=data['query'],
         )
 
-        if not submission_ids:
+        if submission_ids:
+            partial_perms = True
+            data['query'] = {}
+        else:
+            partial_perms = False
             submission_ids = data['submission_ids']
 
         submissions = self.get_submissions(
             user=user,
-            format_type=SUBMISSION_FORMAT_TYPE_JSON,
-            submission_ids=submission_ids
+            format_type=SUBMISSION_FORMAT_TYPE_XML,
+            submission_ids=submission_ids,
+            query=data['query'],
         )
 
-        submission_ids = [int(id_) for id_ in submission_ids]
+        if not self.current_submissions_count:
+            raise KobocatBulkUpdateSubmissionsClientException(
+                detail=_('No submissions match the given `submission_ids`')
```

Comment:
`_()` isn't imported. I'll fix it, but I'll also open a new issue about adding test code that reaches these lines.
Internal discussion: https://chat.kobotoolbox.org/#narrow/stream/4-Kobo-Dev/topic/Unused.20test.20code/near/104338

Reply:
#3784